### ISSUE TITLE
Follow filter issue

### ISSUE THREAD

**Issue by johneakin** [nieroda/mcafeeBTCTradeBot#1]
I am fairly sure that I myself could reply to a tweet  which would trigger  a buy event
The documentation says:
------------------------------------------------------------------------------------------------
follow
A comma-separated list of user IDs, indicating the users whose Tweets should be delivered on the stream. Following protected users is not supported. For each user specified, the stream will contain:

Tweets created by the user.
Tweets which are retweeted by the user.
Replies to any Tweet created by the user.
Retweets of any Tweet created by the user.
Manual replies, created without pressing a reply button (e.g. “@twitterapi I agree”).
------------------------------------------------------------------------------------------------

Not sure if there is a better filter to only stream tweets from the user id itself, otherwise you probably need to wrap the method in an if statement

 if(!tweet.text.includes('@officialmcafee'))
	  {
		checkTweet(tweet.text);
	  }

**Comment by nieroda** [comment]
Thank you for pointing this out. I just fixed this in #2b82fe1

```javascript
 let { in_reply_to_status_id, in_reply_to_status_id_str, in_reply_to_user_id, in_reply_to_user_id_str, in_reply_to_screen_name } = twitter_response;
if (!in_reply_to_status_id && !in_reply_to_status_id_str && !in_reply_to_user_id && !in_reply_to_user_id_str && !in_reply_to_screen_name) {
   return true;
 } else {
   console.log('Tweet is not valid');
   return false;
 }
```

I added a validate function that checks to make sure all of the above values are null, and then it checks to make sure the tweet was made by the correct twitter ID(might have been a given but better to be safe than sorry)

```javascript
tweet.user.id_str === '961445378' //check
```

I have incredibly spotty connection right now so testing takes 10x as long as it should. Please confirm it is fixed and I will close this issue

**Comment by johneakin** [comment]
Seems to fix the issue :)
